src/errorpage.cc

```diff
@@ -1352,6 +1352,11 @@ ErrorState::BuildHttpReply()
             request->detailError(type, detail);
         else
             request->detailError(type, SysErrorDetail::NewIfAny(xerrno));
+    } else if (ale) {
+        if (detail)
+            ale->updateError(Error(type, detail));
+        else
+            ale->updateError(Error(type, SysErrorDetail::NewIfAny(xerrno)));
     }
 
```

Comment:
This small fix does not address some of the Bug 5278 concerns. For example, it does not affect parsing and logging of HTTP headers behind an unexpected relative URL. However, it is a significant improvement because we now inform the admins of what went wrong by logging `%err_code`. The (somewhat controversial) processing/logging of related _details_ can be left to future work/PRs.